src/hooks/useStakes.js

```diff
@@ -0,0 +1,53 @@
+import { useMemo } from 'react'
+import { useAppState } from '../providers/AppState'
+import { useWallet } from '../providers/Wallet'
+
+import { addressesEqual } from '../lib/web3-utils'
+
+// TODO: See how to merge it with the implementation done at app-logic
+export function useMyStakes() {
+  const { account } = useWallet()
+  const { proposals } = useAppState()
+
+  return useMemo(
+    () =>
+      proposals.reduce((acc, proposal) => {
```

Comment:
i think we already have this , you can do ` const {myStakes} = useAppLogic()`

Reply:
Ah yeah but the problem there is that we are also using other hooks that we don't want to use here, for example we are not really interested in the `useProposals` hook which will compute and calculate all conviction values. 

That's why i added a TODO there to see how to reuse it but not use the app-logic hooks that are not needed in this case.

Reply:
Agree on that! let's do it if possible